### PR TITLE
First example of configHolder

### PR DESCRIPTION
**Pull request by oscerd**
Hi Edward,

This is an example PR of what I'm thinking to do about taking properties out of Farsandra class and is related to issue #5 .

If you think this approach can be useful, I'll carry on with the other properties.

Thank you,
Bye.

Andrea

### CODE REVIEW

**Comment by edwardcapriolo** [comment]
I like this. Do you want me to merge this as is?

**Comment by edwardcapriolo** [comment]
Can you please match my formatting. No tabs only 2 space indents?

**Comment by oscerd** [comment]
Hi Edward,

I will match your formatting.

I will update this PR with all the properties I think we can take out.

Thank you,

Bye,
Andrea

**Comment by oscerd** [comment]
I think this last commit is a good PR to start.

I have avoided to take out the paths and properties used in the cassandra.yaml and cassandra-env.sh, I think is not a good solution to put them in properties file.

I think formatting now is good.

Please let me know if you have other plans for this project.

Thanks a lot.
Bye,
Andrea

File: src/test/resources/farsandra.properties [hunk 1]
```diff
@@ -0,0 +1,10 @@
+### Farsandra Properties
+cassandra.package.name.prefix=apache-cassandra-
+cassandra.package.name.suffix=-bin.tar.gz
+cassandra.package.dist.site=http://archive.apache.org/dist/cassandra/
+cassandra.config.file.name=cassandra.yaml
+cassandra.environment.file.name=cassandra-env.sh
+farsandra.home.folder=.farsandra
+farsandra.conf.dir=conf
```
Comment (edwardcapriolo) [inline src/test/resources/farsandra.properties]: We want this to be able to be configured per instance for those running N copies of cassandra in the same test.

**Comment by edwardcapriolo** [comment]
I would like to suggest a few changes. For the properties file you created there is a small issue. You can use farsandra to start N copies of cassandra, and if all instances read the same data directory that will not work.

I have a suggestion. Provide different constructors in the farsandra class.

Farsandra() <- legacy does not read property file
Farsandra(boolean readProps) <- reads default location property file if set to true
Farsandra(File configFile) <- read a specific configuration file.

Than in the code we can do things like this:

```java
if (configHolder == null) {
  File farsandra = new File(home, ".farsandra");
} else {
  File farsandra = new File(home,   this.getConfigHolder().getProperties().getProperty("farsandra.home.folder"));
}
```
I think this gives us the best of both worlds. The classic way continues working and the config file can optionally be used.

**Comment by oscerd** [comment]
You're right.

I've inserted your changes and I think this is what you are you thinking.

Now it's possible to specify different data directory for different instances.

Let me know if everything is ok for you.

Thank you.
Bye,
Andrea

**Pull request merged**